Sources/Customization/DefaultEventProcessor.swift

```diff
@@ -0,0 +1,271 @@
+        static public let maxQueueSize = 10000
+        static let maxFailureCount = 3
+    }
+    
+    // the max failure count.  there is no backoff timer.
+    
+    let eventHandler: OPTEventHandler
+    lazy var logger = OPTLoggerFactory.getLogger()
+    var backingStore: DataStoreType
+    var backingStoreName: String
+    
+    // for dispatching events
+    let dispatcher = DispatchQueue(label: "DefaultEventDispatcherQueue")
+    // using a datastore queue with a backing file
+    let dataStore: DataStoreQueueStackImpl<EventForDispatch>
+    // timer as a atomic property.
+    var timer: AtomicProperty<Timer> = AtomicProperty<Timer>()
+    
+    var observerProjectId: NSObjectProtocol?
+    var observerRevision: NSObjectProtocol?
+    
+    public init(eventHandler: OPTEventHandler = DefaultEventHandler(),
+                batchSize: Int = DefaultValues.batchSize,
+                timerInterval: TimeInterval = DefaultValues.timeInterval,
+                maxQueueSize: Int = DefaultValues.maxQueueSize,
+                backingStore: DataStoreType = .file,
+                dataStoreName: String = "OPTEventQueue") {
+        self.eventHandler = eventHandler
+        self.batchSize = batchSize > 0 ? batchSize : DefaultValues.batchSize
+        self.timerInterval = timerInterval >= 0 ? timerInterval : DefaultValues.timeInterval
+        self.maxQueueSize = maxQueueSize >= 100 ? maxQueueSize : DefaultValues.maxQueueSize
+        
+        self.backingStore = backingStore
+        self.backingStoreName = dataStoreName
+
+        switch backingStore {
+        case .file:
+            self.dataStore = DataStoreQueueStackImpl<EventForDispatch>(queueStackName: "OPTEventQueue",
+                                                                       dataStore: DataStoreFile<[Data]>(storeName: backingStoreName))
+        case .memory:
+            self.dataStore = DataStoreQueueStackImpl<EventForDispatch>(queueStackName: "OPTEventQueue",
+                                                                       dataStore: DataStoreMemory<[Data]>(storeName: backingStoreName))
+        case .userDefaults:
+            self.dataStore = DataStoreQueueStackImpl<EventForDispatch>(queueStackName: "OPTEventQueue",
+                                                                       dataStore: DataStoreUserDefaults())
+        }
+        
+        if self.maxQueueSize < self.batchSize {
+            self.logger.e(.eventDispatcherConfigError("batchSize cannot be bigger than maxQueueSize"))
+            self.maxQueueSize = self.batchSize
+        }
+        
+        addProjectChangeNotificationObservers()
+        
+        subscribe()
+    }
+    
+    deinit {
+        stopTimer()
+        
+        removeProjectChangeNotificationObservers()
+
+        unsubscribe()
+    }
+    
+    open func process(event: BatchEvent, completionHandler: DispatchCompletionHandler? = nil) {
+        guard dataStore.count < maxQueueSize else {
```

Comment:
no flush?

Reply:
Not clear what you mean here. That line is for overflow dropping and flush on batch size is handled below. Can you clarify if you mean something else?